vignettes/writing-and-reviewing-text.Rmd

```diff
@@ -0,0 +1,156 @@
+8. Anything not explicitly mentioned may or may not work; you can try it out and update this guide!
+
+## Automatic conversion
+
+### Word to markdown...
+
+The conversion makes use of the packages [`{officer}`](https://davidgohel.github.io/officer/) and [`{xml2}`](https://xml2.r-lib.org/), along with some common tidyverse packages.
+
+Running `word_to_md()` will, for each 'chunk' in the document
+
+1. Get plain text on each line
+2. Use the [xml](https://en.wikipedia.org/wiki/XML) that underlies the Word doc to create a map of the various styles and hyperlinks
+3. Transform the plain text using the style maps, into a a version that uses markdown syntax
+4. Write the final text to a markdown file
+
+### ...markdown to Word
+
+Conversion from markdown to Word is much easier. Running `md_to_word()` will, for each markdown file
+
+1. Read each line into a list
+2. Add the file path to the beginning of each list
+3. Combine all lists into one list
+
+This list is then written as an Rmarkdown document, and finally rendered as a Word doc using the [`rmarkdown::render`](https://pkgs.rstudio.com/rmarkdown/reference/render.html) and [`officedown::rdocx_document`](https://davidgohel.github.io/officedown/reference/rdocx_document.html) functions. The style document specified by `review/styles/draft-styles.Rmd` is also generated and used in the render step.
+
+## Assisted review
+
+Once the text is first added to the app, as markdown files, there is probably going to be a time when something needs to change, based on feedback or a rethink of what is written. Using the automation provided by the review functions makes keeping the draft text and the app text synced quicker, easier and less error-prone.
+
+The end-to-end process follows.
+
+1. Generate the initial Word document by running `gen_review_doc()`. The Word doc will be saved in the `inst/review` folder as `review.docx`.
+2. Write the initial draft of text in `review.docx`, using Word's review functionality to track changes and comments. It is important to use the conventions laid out in ['Word-first' approach](127.0.0.1/#word-first-approach), as without these the functions may give unexpected results. Keep it in Word for as long as possible!
+3. When you get to a point when you _have_ to deploy the app with the text, run `word_to_md()`. With the default arguments, the generated markdown files are saved to `inst/review/temp`. From this point on, the app text and the Word text need to be synced as changes are agreed. On the first run of `word_to_md` the initial snapshot files will be automatically created in `inst/review/tests/_snaps/review_md`. These snapshots will be used to compare future changes to, and can be updated as the text changes and is added to the app. Note that if any existing markdown is in `inst/review/tests/_snaps/review_md` folder, the initial snaps will not be created, so ensure the folder is either not present or empty.
```

Comment:
Note that if any existing markdown is in `inst/review/tests/_snaps/review_md` folder, the initial snaps will not be created, so ensure the folder is either not present or empty. -> wonder add warning if _snaps folders already exist?

Reply:
https://github.com/nhsbsa-data-analytics/nhsbsaShinyR/pull/27/files/729bd880101e4c2092046400a3e9c9d7cfb635df#diff-2afd1b733e88398171623e4be07eae000888866447d3d7be810d50974ecdc114R627-R639

Does this handle it well enough? I will add a note about the option in vignette.

Reply:
Added warning after all and edited vignette to explain.